solana/indexer/balance_changes.go

```diff
@@ -0,0 +1,152 @@
+package indexer
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"slices"
+
+	"bridgerton.audius.co/database"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/jackc/pgx/v5"
+	"go.uber.org/zap"
+)
+
+func processBalanceChanges(
+	ctx context.Context,
+	db database.DBTX,
+	slot uint64,
+	meta *rpc.TransactionMeta,
+	tx *solana.Transaction,
+	blockTime time.Time,
+	trackedMints []string,
+	txLogger *zap.Logger,
```

Comment:
I so badly want to suggest args objects instead of positional, but I don't have a good reason other than personal preference :-).

Reply:
yeah i don't know what's "idiomatic" for go. I like object args too but I feel like it might litter the code base with tons of extra structs? the lack of "exports" to keep things private to a file makes it feel like it actually has a cost